Documentation of coq_makefile: fix name of installation dir + help on using option -f (#12884)
coq/coq: doc/sphinx/practical-tools/utilities.rst

Such command generates the following files:

CoqMakefile
  is a makefile for ``GNU Make`` with targets to build the project
  (e.g. generate .vo or .html files from .v or compile .ml* files)

>   and install it in the ``user-contrib`` directory where |Coq|

Reviewer: ```suggestion
  and install it in the ``user-contrib`` directory where the |Coq|
```

The rest is fine.

CoqMakefile

>   is a generic makefile for ``GNU Make`` that provides

Reviewer: ```suggestion
  a generic makefile for ``GNU Make`` that provides
```

>   is a generic makefile for ``GNU Make`` that provides
>   targets to build the project (both ``.v`` and ``.ml*`` files), to install it

Reviewer: ```suggestion
  a makefile for ``GNU Make`` with
  targets to build the project (both ``.v`` and ``.ml*`` files), install it
```

Author: Since the next item has the form of a sentence: "CoqMakefile.conf contains", I would keep the "is", wouldn't I? I would otherwise adopt the light form of the suggestion.

A side question is why is it better to call it `CoqMakefile`, rather than just `Makefile`?

Reviewer: > Since the next item has the form of a sentence: "CoqMakefile.conf contains", I would keep the "is", wouldn't I?

It's not necessary to have complete sentences with a list of items, especially when the item names ("CoqMakefile", etc.) are on separate lines.  Nor is it necessary to be fully consistent with "contains".  Suppose you have a list of a dozen items.  It is repetitive and uninteresting to repeat "is" on every single one of them.  (The "Such command" above could be removed for the same reason; you really don't need to see "This tactic ..."  or "This command ..." beginning the description of hundreds of those constructs.)  

Is French stricter about everything being full sentences?

I leave it up to you.

> A side question is why is it better to call it `CoqMakefile`, rather than just `Makefile`?

An excellent point.  It would simplify usage and the doc a little.

>   system-wide (i.e. in the ``user-contrib`` subdirectory where |Coq| is installed)
>   as well as to invoke coqdoc to generate HTML documentation. Note:
>   pass option ``-f CoqMakefile`` to ``make`` to use the so-generated
>   file ``CoqMakefile``.

Reviewer: ```suggestion
  in the ``lib/user-contrib`` directory of the |Coq| installation
  and invoke ``coqdoc`` to generate HTML documentation.
  Run ``make`` with the ``-f CoqMakefile`` option to use ``CoqMakefile``.
```

The `coqdoc` output goes into `doc` directory of the Coq installation?  Maybe worth mentioning that.

Thoughts beyond the scope of this PR:

CoqIDE doesn't provide an easy way to navigate to the `coqdoc` (e.g. on the `Help` menu.  That may be worth fixing.

The Coq installation directory name doesn't include "coq" for opam.  That seems odd but maybe it's OK:

```
$ which coqtop
/home/jfehrle/.opam/4.06.0/bin/coqtop
$ find . -name user-contrib
./lib/coq/user-contrib
./.opam-switch/sources/coq.8.11.2/user-contrib
```

Author: > The  `coqdoc` output goes into `doc` directory of the Coq installation? Maybe worth mentioning that.

At first in the local directory. Then in the installation directory if `make install` is called. Somehow, the whole sentence could be reformulated into:

```
is a makefile for ``GNU Make`` with targets to build the project (e.g. ``.vo`` 
and ``.html` from ``.v`` and ``.cmx`` or ``.cmo`` from ``.ml*`` files, among 
others) and to install it in the ``user-contrib`` subdirectory of the |Coq| installation.
Run ``make`` with the ``-f CoqMakefile`` option to use ``CoqMakefile``.
```

> CoqIDE doesn't provide an easy way to navigate to the coqdoc

What do you mean? To browse the html possibly associated to the current buffer? Indeed, that would be nice to have. But see also what Clément Pit-Claudel is doing around [literate proving](https://plv.csail.mit.edu/blog/drafts/alectryon.html).

> Why is the `menhir` documentation in the Coq installation? It seems irrelevant. It should go away, right?

What do you mean?

Reviewer: > CoqIDE doesn't provide an easy way to navigate to the coqdoc

We have Help/Browse Coq Library which IIUC is coqdoc output.  I was thinking of something similar for locally-built coqdoc, not tying it to the buffer name (though that would be neat).

You don't have a `doc/menhir/manual.pdf` in your official installation (assuming you use Linux)? 
 `menhir` documentation may only be present in the Windows build.  I'll open an issue about that separately.

> is a makefile for ``GNU Make`` with targets to build the project (e.g. ``.vo`` 
and ``.html` from ``.v`` and ``.cmx`` or ``.cmo`` from ``.ml*`` files, among 
others) and to install it in the ``user-contrib`` subdirectory of the |Coq| installation.
Run ``make`` with the ``-f CoqMakefile`` option to use ``CoqMakefile``.

You don't want to say `lib/user-contrib` here?

The part in parentheses is wordy.  I think "``.cmx`` or ``.cmo`` from ``.ml*`` files" doesn't need to be stated.  Maybe it could just be "... build the project (e.g. generate `.vo` files from `.v` and compile `.ml*` files), generate `coqdoc`, ...".  Having both "e.g." and "among others" is redundant.

`coqdoc` claims to support more than just html output.  Does `CoqMakefile` only support html?

In English, "to verb1 ..., to verb2 ... and to verb3" can be written either that way or simply as "to verb1 ..., verb2 ... and verb3" provided all the verbs are infinitives.  The longer form perhaps emphasizes the verbs a bit more.

I'll leave it up to you.

Author: > > CoqIDE doesn't provide an easy way to navigate to the coqdoc
> 
> We have Help/Browse Coq Library which IIUC is coqdoc output. I was thinking of something similar for locally-built coqdoc, not tying it to the buffer name (though that would be neat).

That would be nice, yes.

> You don't want to say `lib/user-contrib` here?

On Linux/MacOS, it is is in `lib/coq/user-contrib`. It is `lib/user-contrib` only on Windows I believe. I used a periphrase.

> The part in parentheses is wordy. I think "`.cmx` or `.cmo` from `.ml*` files" doesn't need to be stated. Maybe it could just be "... build the project (e.g. generate `.vo` files from `.v` and compile `.ml*` files), generate `coqdoc`, ...". Having both "e.g." and "among others" is redundant.

OK.

> `coqdoc` claims to support more than just html output. Does `CoqMakefile` only support html?

It supports also `tex` output.

> In English, "to verb1 ..., to verb2 ... and to verb3" can be written either that way or simply as "to verb1 ..., verb2 ... and verb3" provided all the verbs are infinitives. The longer form perhaps emphasizes the verbs a bit more.
> 
> I'll leave it up to you.

OK, I pushed a new version synthesizing what we said.
